docs/data-model.md

```
@@ -0,0 +1,349 @@
+# Data Model & Relationships
+
+## Overview
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)